File: Solutions/IQCare.Core/IQCare.PMTCT.BusinessProcess/Commands/HaartProphylaxisCommand.cs
using IQCare.Library;
using IQCare.PMTCT.Core.Models;
using MediatR;
using System.Collections.Generic;

namespace IQCare.PMTCT.BusinessProcess.Commands
{
    public class HaartProphylaxisCommand :IRequest<Result<HaartProphylaxisResponse>>
    {
        public List<PatientDrugAdministration> PatientDrugAdministration;
        public List<PatientChronicIllness> PatientChronicIllnesses;
        public int OtherIllness { get; set; }
    }

    public class HaartProphylaxisResponse
    {
        public int PreventivePartnerId { get; set; }
    }
}

File: Solutions/IQCare.Core/IQCare/frontend/app/pmtct/_models/HaartProphylaxisCommand.ts
import {PatientDrugAdministration} from './PatientDrugAdministration';
import {PatientChronicIllness} from './PatientChronicIllness';

export interface HaartProphylaxisCommand {
    PatientDrugAdministration: PatientDrugAdministration[];
    PatientChronicIllnesses: PatientChronicIllness[];
    OtherIllness?: number;
}
